standalone/portal-api/src/modules/company-information/services/supplier-information.service.ts

```diff
@@ -0,0 +1,160 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { RelationshipService } from 'src/modules/relationship/services/relationship.service';
+
+import { BaseService } from 'src/common/services/base.service';
+import { IsolationLevel, Transactional } from 'src/common/transaction';
+import { SupplierInformationResponse } from '../dto/response/supplier/get-supplier-response.dto';
+import { CreateSupplierInformationRequest } from '../dto/request/supplier/create-supplier-company.dto';
+import { SupplierCompanyEntityService } from 'src/modules/company/services/supplier-entity.service';
+import { QuickApplicationEntity } from 'src/modules/applications/models/quickapplication.entity';
+import { BuyerQuickApplicationEntityRepository } from 'src/modules/applications/repositories/buyer.quickapplication.repository';
+import { EntityMappingService } from './mapping.service';
+import { UpdateSupplierInformationRequest } from '../dto/request/supplier/update-supplier-company.dto';
+import { BuyerCompanyEntity } from 'src/modules/company/models/buyer.company.entity';
+import { BuyerCompanyEntityRepository } from 'src/modules/company/repositories/buyer.company.repository';
+import { triggerError } from 'src/common/trigger.error';
+
+@Injectable()
+export class SupplierInformationService extends BaseService {
+  constructor(
+    @InjectRepository(QuickApplicationEntity)
+    private buyerApplicationRepository: BuyerQuickApplicationEntityRepository,
+
+    @InjectRepository(BuyerCompanyEntity)
+    private buyerCompanyRepository: BuyerCompanyEntityRepository,
+
+    private readonly mappingService: EntityMappingService,
+    private readonly supplierService: SupplierCompanyEntityService,
+    private readonly relationshipService: RelationshipService,
+  ) {
+    super();
+  }
+
+  public async getSupplierInformation(
+    applicationId: string,
+  ): Promise<SupplierInformationResponse> {
+    const application = await this.buyerApplicationRepository.findOne(
+      applicationId,
+      {
+        relations: ['supplierInformation'],
+      },
+    );
+
+    const savedSupplier = await this.supplierService.getSupplierEntity(
+      application.supplierInformation.id,
+    );
+
+    const savedRelationship = await this.relationshipService.getEntity(
+      application.company.id,
+      application.supplierInformation.id,
+    );
+
+    return this.mappingService.mapSupplierInformationForImporterApplication(
+      savedSupplier.id,
+      savedRelationship,
+    );
+  }
+
+  @Transactional({
+    isolationLevel: IsolationLevel.READ_COMMITTED,
+  })
+  public async createSupplierInformation(
+    data: CreateSupplierInformationRequest,
+    applicationId: string,
+  ): Promise<SupplierInformationResponse> {
+    this.setupTransactionHooks();
+
+    //TODO: check for application so that isolated supplier creation is prevented
```

Comment:
What do you mean?

Reply:
This comment actually got copied from the previous code when i created this service, the TODO was about making sure duplicate supplier is not creating while a buyer fills in supplier details which was the case prior to this PR. I have removed the comment